test/53-assert-10-assert_link_exist.bats

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bats
+
+load 'test_helper'
+fixtures 'exist'
+
+# Correctness
+@test 'assert_link_exist() <file>: returns 0 if <file> Link exists' {
+  local -r file="${TEST_FIXTURE_ROOT}/dir/link"
+  run assert_link_exist "$file"
+  [ "$status" -eq 0 ]
+  [ "${#lines[@]}" -eq 0 ]
+}
+
+@test 'assert_link_exist() <file>: returns 1 and displays path if <file>symbolic link does not exist' {
+  local -r file="${TEST_FIXTURE_ROOT}/dir/ "
+  run assert_link_exist "$file"
+  [ "$status" -eq 1 ]
+  [ "${#lines[@]}" -eq 3 ]
+  [ "${lines[0]}" == '-- symbolic link does not exist --' ]
+  [ "${lines[1]}" == "path : $file" ]
+  [ "${lines[2]}" == '--' ]
+}
+
+# Transforming path
+@test 'assert_link_exist() <file>: replace prefix of displayed path' {
+  local -r BATSLIB_FILE_PATH_REM="#${TEST_FIXTURE_ROOT}"
+  local -r BATSLIB_FILE_PATH_ADD='..'
+  run assert_link_exist "${TEST_FIXTURE_ROOT}/nodir"
+  [ "$status" -eq 1 ]
+  [ "${#lines[@]}" -eq 3 ]
+  [ "${lines[0]}" == '-- symbolic link does not exist --' ]
+  [ "${lines[1]}" == "path : ../nodir" ]
+  [ "${lines[2]}" == '--' ]
+}
+
+@test 'assert_link_exist() <file>: replace suffix of displayed path' {
+  local -r BATSLIB_FILE_PATH_REM='%file.does_not_exist'
+  local -r BATSLIB_FILE_PATH_ADD='..'
+  run assert_link_exist "${TEST_FIXTURE_ROOT}/nodir"
+  [ "$status" -eq 1 ]
+  [ "${#lines[@]}" -eq 3 ]
+  [ "${lines[0]}" == '-- symbolic link does not exist --' ]
+  [ "${lines[1]}" == "path : ${TEST_FIXTURE_ROOT}/nodir" ]
```

Comment:
if you first would remove "nodir" from the path, then you would need here to test `[ "${lines[1]}" == "path : ${TEST_FIXTURE_ROOT}/.." ]`